docs/static/dead-letter-queues.asciidoc

```diff
@@ -0,0 +1,155 @@
+
+[source,yaml]
+-------------------------------------------------------------------------------
+dead_letter_queue.enable: true
+-------------------------------------------------------------------------------
+
+Dead letter queues are stored as files in the local directory of the Logstash
+instance. By default, the dead letter queue files are stored in
+`path.data/dead_letter_queue`. Each pipeline has a separate queue. For example,
+the dead letter queue for the `main` pipeline is stored in
+`LOGSTASH_HOME/data/dead_letter_queue/main` by default. The queue files are
+numbered sequentially: `1.log`, `2.log`, and so on.
+
+You can set `path.dead_letter_queue` in the `logstash.yml` file to
+specify a different path for the files:
+
+[source,yaml]
+-------------------------------------------------------------------------------
+path.dead_letter_queue: "path/to/data/dead_letter_queue"
+-------------------------------------------------------------------------------
+
+===== File Rotation
+
+Dead letter queues have a built-in file rotation policy that manages the file
+size of the queue. When the file size reaches a preconfigured threshold,  a new
+file is created automatically. The size limit of the dead letter queue is
+constrained only by the amount of space that you have available on disk.
+
+NOTE: Dead letter queues retain all the events that are written to them.
+Currently, you cannot configure the size of the queue or the size of the files
+that are used to store the queue. 
+
+//REVIEWIERS: I feel that we have to say something about this ^^ because users will wonder, but I'm not sure what we should say wrt future plans to make this configurable.
+
+[[processing-dlq-events]]
+==== Processing Events in the Dead Letter Queue
+
+When you are ready to process events in the dead letter queue, you create a
+pipeline that uses the `dead_letter_queue` input plugin to read from the dead
+letter queue. The pipeline configuration that you use depends, of course, on
+what you need to do. For example, if the dead letter queue contains events that
+resulted from a mapping error in Elasticsearch, you can create a pipeline that
+reads the "dead" events, removes the field that caused the mapping issue, and
+re-indexes the clean events into Elasticsearch. 
+
+The following example shows how to read events from the dead letter queue and
+write the events to standard output: 
+
+[source,yaml]
+--------------------------------------------------------------------------------
+input {
+  dead_letter_queue {
+    path => "/path/to/data/dead_letter_queue" <1>
+    commit_offsets => true <2>
+    pipeline_id => "main" <3>
+  }
+}
+
+output {
+  stdout {
+    codec => rubydebug
+  }
+}
+--------------------------------------------------------------------------------
+
+<1> The path to the directory containing the dead letter queue. This is either
+the default, `path.data/dead_letter_queue`, or the value specified for
+`path.dead_letter_queue` in the `logstash.yml` file. 
```

Comment:
I'll add something about this and see how it works. The main point here is to help users understand how they can find out where the files reside.